File: solutions/typescript/transpose/1/transpose.ts
export function transpose(matrix: string[]): string[] {
    const result: string[] = [];

    // Находим максимальную длину строки в матрице
    const maxLen = Math.max(0, ...matrix.map(row => row.length));

    for (let col = 0; col < maxLen; col++) {
        let currentTrailingSpaces = '';
        let transposedRow = '';

        for (let row = 0; row < matrix.length; row++) {
            // Если символ существует, добавляем его
            if (col < matrix[row].length) {
                // Если до этого накопились пробелы, сначала добавляем их
                transposedRow += currentTrailingSpaces + matrix[row][col];
                currentTrailingSpaces = '';
            } else {
                // Если символа нет, временно откладываем пробел
                currentTrailingSpaces += ' ';
            }
        }
        
        result.push(transposedRow);
    }

    return result;
}

File: solutions/kotlin/transpose/1/src/main/kotlin/Transpose.kt
object Transpose {
    fun transpose(input: List<String>): List<String> {
        if (input.isEmpty()) return emptyList()


        val maxCols = input.maxOf { it.length }
        val result = mutableListOf<String>()


        for (col in 0 until maxCols) {
            val newRow = StringBuilder()

       
            for (row in input.indices) {
                if (col < input[row].length) {
                    newRow.append(input[row][col])
                } else {
                    newRow.append(' ')
                }
            }
            
            result.add(newRow.toString())
        }

        val finalResult = mutableListOf<String>()
        for (i in result.indices) {

            var maxRequiredLength = 0
            for (j in input.indices) {
                if (input[j].length > i) {
                    maxRequiredLength = j + 1
                }
            }
            finalResult.add(result[i].substring(0, maxRequiredLength))
        }

        return finalResult
    }
}
